src/test/java/hudson/plugins/active_directory/ActiveDirectoryDomainFipsEnabledTest.java

```diff
@@ -0,0 +1,78 @@
+import io.jenkins.plugins.casc.ConfiguratorException;
+import org.jvnet.hudson.test.recipes.LocalData;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+import static org.junit.Assert.fail;
+import static org.jvnet.hudson.test.LoggerRule.recorded;
+
+public class ActiveDirectoryDomainFipsEnabledTest {
+
+    @Rule
+    public JenkinsRule jenkinsRule = new JenkinsRule();
+
+    @Rule
+    public LoggerRule loggerRule = new LoggerRule().record("hudson.diagnosis.OldDataMonitor", Level.INFO).capture(1000);
+
+    @ClassRule
+    public static TestRule fip140Prop = FlagRule.systemProperty("jenkins.security.FIPS140.COMPLIANCE", "true");
+
+    @Test
+    public void smokeTests() {
+
+        ActiveDirectoryDomain.DescriptorImpl adDescriptor = ExtensionList.lookupSingleton(ActiveDirectoryDomain.DescriptorImpl.class);
+
+        // error message should be displayed if a FIPS non-compliant option is chosen
+        FormValidation resultError = adDescriptor.doCheckTlsConfiguration(TlsConfiguration.TRUST_ALL_CERTIFICATES.name());
+        assertEquals("Insecure TLS configuration should not be allowed", FormValidation.Kind.ERROR, resultError.kind);
+
+        // if a FIPS compliant option is chosen, no error message should be displayed
+        FormValidation resultOk = adDescriptor.doCheckTlsConfiguration(TlsConfiguration.JDK_TRUSTSTORE.name());
+        assertEquals("Secure TLS configuration should be allowed", FormValidation.Kind.OK, resultOk.kind);
+
+        assertThrows("Insecure TLS configuration should not be allowed", IllegalArgumentException.class,
+                     () -> new ActiveDirectoryDomain("name", "the_server", "site", "bindName", "bindPassword", TlsConfiguration.TRUST_ALL_CERTIFICATES));
+
+        try {
+            new ActiveDirectoryDomain("name", "the_server", "site", "bindName",
+                                                                     "bindPassword", TlsConfiguration.JDK_TRUSTSTORE);
+        } catch (Exception e) {
+            fail("Secure TLS configuration should be allowed");
+        }
+    }
+
+    @Test
+    public void cascTest() {
+        assertThrows("Insecure TLS configuration should not be allowed", ConfiguratorException.class,
+                     () -> ConfigurationAsCode.get().configure(Paths.get("src/test/resources/hudson/plugins/active_directory/ActiveDirectoryDomainFipsEnabledTest/configuration-as-code-insecure.yaml").toString()));
+        try {
+            ConfigurationAsCode.get().configure(Paths.get("src/test/resources/hudson/plugins/active_directory/ActiveDirectoryDomainFipsEnabledTest/configuration-as-code-secure.yaml").toString());
+        } catch (Exception e) {
+            fail("Secure TLS configuration should be allowed");
+        }
+    }
+
+    @Test
+    @LocalData
+    public void testBlowsUpOnStart() throws Throwable {
```

Comment:
I guess @rsandell 's comment about the configRoundtrip means not only this test. It's about having a LocalData with a valid configuration and then call https://github.com/jenkinsci/jenkins-test-harness/blob/0e840c415bfa9fc16b380a74275666da356ef773/src/main/java/org/jvnet/hudson/test/JenkinsRule.java#L1433-L1438 (`jenkinsRule.configRoundtrip()`)

Reply:
I mean trying to `form.submit()` with an invalid configuration and see an Angry Jenkins judging you :)

Reply:
mentioning configRoundtrip was unfortunately a bit misleading wording from me, it was before I realized that you can't really do a traditional configRoundtrip test because you can't create the bad backend config object to roundtrip on :)
You need to create one that is OK according to FiPS and then change the form value and submit.